apps/laboratory/tests/shared/fixtures/w3m-wallet-fixture.ts

```diff
@@ -17,81 +16,73 @@ interface ModalWalletFixture {
-      const uri = await modalPage.getConnectUri(timingRecords)
-      timeEnd('modalPage.getConnectUri')
-      timeStart('walletPage.connectWithUri')
-      await walletPage.connectWithUri(uri)
-      timeEnd('walletPage.connectWithUri')
-      const connectionInitiated = new Date()
-      timeStart('walletPage.handleSessionProposal')
-      await walletPage.handleSessionProposal(DEFAULT_SESSION_PARAMS)
-      timeEnd('walletPage.handleSessionProposal')
-      const proposalReceived = new Date()
-      timingRecords.push({
-        item: 'receiveSessionProposal',
-        timeMs: proposalReceived.getTime() - connectionInitiated.getTime()
-      })
-      timeStart('walletValidator.expectConnected')
-      await walletValidator.expectConnected()
-      timeEnd('walletValidator.expectConnected')
-      await use(walletPage)
-    } else {
-      timeStart('doActionAndWaitForNewPage')
-      const page = await doActionAndWaitForNewPage(modalPage.clickWalletDeeplink(), context)
-      let pairingCreatedTime: Date | null = null
-      let verificationStartedTime: Date | null = null
-      page.on('console', msg => {
-        if (msg.text().includes('set') && msg.text().includes('core/pairing/pairing')) {
-          pairingCreatedTime = new Date()
-        }
-        if (msg.text().includes('resolving attestation')) {
-          verificationStartedTime = new Date()
-        }
-        if (msg.text().includes('session_proposal') && msg.text().includes('verifyContext')) {
-          // For some reason this log is emitted twice; so only recording the time once
-          if (verificationStartedTime) {
-            const verificationEndedTime = new Date()
-            timingRecords.push({
-              item: 'sessionProposalVerification',
-              timeMs: verificationEndedTime.getTime() - verificationStartedTime.getTime()
-            })
-            verificationStartedTime = null
-          }
+    // Setup
+    let pairingCreatedTime: Date | null = null
+    let verificationStartedTime: Date | null = null
+
+    timeStart('new WalletPage')
+    const walletPage = new WalletPage(await context.newPage())
+    timeEnd('new WalletPage')
+
+    walletPage.page.on('console', msg => {
+      if (msg.text().includes('set') && msg.text().includes('core/pairing/pairing')) {
+        pairingCreatedTime = new Date()
+      }
+      if (msg.text().includes('resolving attestation')) {
+        verificationStartedTime = new Date()
+      }
+      if (msg.text().includes('session_proposal') && msg.text().includes('verifyContext')) {
+        // For some reason this log is emitted twice; so only recording the time once
+        if (verificationStartedTime) {
+          const verificationEndedTime = new Date()
+          timingRecords.push({
+            item: 'sessionProposalVerification',
+            timeMs: verificationEndedTime.getTime() - verificationStartedTime.getTime()
+          })
+          verificationStartedTime = null
```

Comment:
@chris13524 can you double check this? 
Code seemed completely duplicated between solana and non-solana after the qr uri change so I merged it. 
Only difference wat this event listener on the console that was only set on the non-solana side.

Reply:
The canary doesn't run Solana tests, so if this is just in the non-Solana variant that's fine

Reply:
There are no variants now, like this runs on every wallet test. If we need to scope it so that it doesnt add the event handler on solana I can add that check

Reply:
That's not a problem. This can run for Solana. It just measures time and adds to the array of time entries. The value is only read in the canary spec